### PR TITLE
DAOS-6261 ci: Improve checkpatch to only inspect modified files.

### PR DESCRIPTION
**Pull request by ashleypittman**
Rather than test the entire tree, only check files that are
touched by the PR.

Fix spelling mistake.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
This cuts the test time by about eight minutes, obviously less if more files are touched.

**Review by sylviachanoiyee — DISMISSED** [review]

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: ci/patch_src_in_place [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 # shellcheck disable=SC2046,SC2035
-codespell -w --ignore-words-list dedup,nd,uint,ths,ba,creat,te,cas,mapp,pres,crashers,dout,tre,reord,mimick,cloneable,keypair,bject,tread,cancelled,controllerd --builtin clear,rare,informal,names,en-GB_to_en-US --skip *.png,*.PNG,*.pyc,src/rdb/raft/*,src/control/vendor/*,RSA.golden,utils/rpms/* $(git ls-tree --full-tree --name-only HEAD)
+codespell -w --ignore-words-list dedup,nd,uint,ths,ba,creat,te,cas,mapp,pres,crashers,dout,tre,reord,mimick,cloneable,keypair,bject,tread,cancelled,controllerd --builtin clear,rare,informal,names,en-GB_to_en-US --skip *.png,*.PNG,*.pyc,src/rdb/raft/*,src/control/vendor/*,RSA.golden,utils/rpms/* $(git diff --name-only origin/"$CHANGE_TARGET" .)
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply ci/patch_src_in_place]: You're quite right.  I've changed this a bit and added the -x setting so we can see what it's checking.  There are more recent master commits so with this change it should be possible to verify it's checking the correct files.

[the remark this answers is not in the document]
Comment (ashleypittman) [reply ci/patch_src_in_place]: This looks to have worked correctly according to the logs.

File: ci/patch_src_in_place [hunk 2]
```diff
@@ -1,7 +1,9 @@
 #!/bin/sh
 
+set -x
+
 # shellcheck disable=SC2046,SC2035
-codespell -w --ignore-words-list dedup,nd,uint,ths,ba,creat,te,cas,mapp,pres,crashers,dout,tre,reord,mimick,cloneable,keypair,bject,tread,cancelled,controllerd --builtin clear,rare,informal,names,en-GB_to_en-US --skip *.png,*.PNG,*.pyc,src/rdb/raft/*,src/control/vendor/*,RSA.golden,utils/rpms/* $(git ls-tree --full-tree --name-only HEAD)
+codespell -w --ignore-words-list dedup,nd,uint,ths,ba,creat,te,cas,mapp,pres,crashers,dout,tre,reord,mimick,cloneable,keypair,bject,tread,cancelled,controllerd --builtin clear,rare,informal,names,en-GB_to_en-US --skip *.png,*.PNG,*.pyc,src/rdb/raft/*,src/control/vendor/*,RSA.golden,utils/rpms/* $(git diff --name-only origin/"$CHANGE_TARGET"...)
```
Comment (brianjmurrell) [inline ci/patch_src_in_place:6]: Interesting.  I have not seen this method of determining the patch against it's base.  It seems to work.

**Review by brianjmurrell — APPROVED** [review]

**Review by sylviachanoiyee — APPROVED** [review]

File: ci/patch_src_in_place (hunk 2 above)
Comment (ashleypittman) [reply ci/patch_src_in_place:6]: What's unique to doing this under CI is performing it against origin/, that took some time to work out.

**Pull request merged**